be/src/formats/parquet/column_reader.cpp

```diff
@@ -167,9 +168,85 @@ class ListColumnReader : public ColumnReader {
     std::unique_ptr<ColumnReader> _element_reader;
 };
 
+class MapColumnReader : public ColumnReader {
+public:
+    explicit MapColumnReader(const ColumnReaderOptions& opts) : _opts(opts) {}
+    ~MapColumnReader() override = default;
+
+    Status init(const ParquetField* field, std::unique_ptr<ColumnReader> key_reader,
+                std::unique_ptr<ColumnReader> value_reader) {
+        _field = field;
+        _key_reader = std::move(key_reader);
+        _value_reader = std::move(value_reader);
+        return Status::OK();
+    }
+
+    Status prepare_batch(size_t* num_records, ColumnContentType content_type, vectorized::Column* dst) override {
+        vectorized::NullableColumn* nullable_column = nullptr;
+        vectorized::MapColumn* map_column = nullptr;
+        if (_field->is_nullable) {
+            DCHECK(dst->is_nullable());
+            nullable_column = down_cast<vectorized::NullableColumn*>(dst);
+            DCHECK(nullable_column->mutable_data_column()->is_map());
+            map_column = down_cast<vectorized::MapColumn*>(nullable_column->mutable_data_column());
+        } else {
+            DCHECK(dst->is_map());
+            map_column = down_cast<vectorized::MapColumn*>(dst);
+        }
+        auto* key_column = map_column->keys_column().get();
+        auto st = _key_reader->prepare_batch(num_records, content_type, key_column);
+        if (!st.ok() && !st.is_end_of_file()) {
+            return st;
+        }
+
+        auto* value_column = map_column->values_column().get();
+        st = _value_reader->prepare_batch(num_records, content_type, value_column);
```

Comment:
do we need to check if num of `key_column` equals to num of `value_column`

Reply:
I think parquet will check and promise it. I will add DCHECK used for debug